api/base/serializers.py

```diff
@@ -14,17 +14,21 @@ def _rapply(d, func, *args, **kwargs):
     else:
         return func(d, *args, **kwargs)
 
+
 def _url_val(val, obj, serializer, **kwargs):
     """Function applied by `HyperlinksField` to get the correct value in the
     schema.
     """
     if isinstance(val, Link):  # If a Link is passed, get the url value
         return val.resolve_url(obj, **kwargs)
-    elif isinstance(val, basestring):  # if a string is passed, it's a function of the serializer
+    elif isinstance(val, basestring):  # if a string is passed, it's an attribute
         return getattr(serializer, val)(obj)
+    elif isinstance(val, WaterbutlerLink):  # If a WaterbutlerLink is passed, get the url value
```

Comment:
Might make sense for `WaterbutlerLink` to subclass `Link`. This conditional would be unnecessary.